active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+
+![Custom UI](customuiexample.png)
+
+```xaml
+<TextBox 
+        x:Name="Name" 
+        Header="Name: " 
+        PlaceholderText="Enter Name..." 
+
+        Text="{x:Bind   ViewModel.Person.Name, 
+                        UpdateSourceTrigger=PropertyChanged, 
+                        Mode=TwoWay}" 
+         UseSystemValidationVisuals="False"
+        />
+
+<!-- Custom error message template -->
+<ItemsControl
+    ItemsSource="{x:Bind Name.(Validation.Errors), Mode=OneWay}"
+    Foreground="Red">
+
+    <ItemsControl.ItemTemplate>
+        <DataTemplate x:DataType="ValidationError">
+            <StackPanel Orientation="Horizontal" Spacing="5">
+                <Border Background="Red" Margin="3" CornerRadius="10" 
+                        Width="12" Height="12" VerticalAlignment="Top">
+                    <TextBlock Text="!" Foreground="White" 
+                               HorizontalAlignment="Center" VerticalAlignment="Center"/>
+                </Border>
+                <TextBlock Text="{x:Bind ErrorMessage, Mode=OneWay}"
+                           />
+            </StackPanel>
+        </DataTemplate>
+    </ItemsControl.ItemTemplate>
+</ItemsControl>
+```
+
+## Customizing the error template
+
+To modify the styling and content of your error template, use the Validation.ErrorTemplate property. The Validation.ErrorTemplate refers to either the content of the tooltip or the content of the container below the control depending on the validation mode.
+
+If you wish to override the default visuals for errors across all forms elements in your app, override the ValidationErrorTemplate DataTemplate and use in your app.
+
+```xaml
+<ResourceDictionary>
+    <DataTemplate x:Key="ValidationErrorTemplate">
+        <StackPanel Orientation="Horizontal">
+            <Border Background="Red" Margin="3" CornerRadius="10" Width="20" Height="20" VerticalAlignment="Top">
+                <TextBlock Text="!" Foreground="White"
+                HorizontalAlignment="Center" VerticalAlignment="Center"/>
+            </Border>
+            <TextBlock Text="{x:Bind (Validation.Errors)[0].ErrorContent, Mode=OneWay}"
+        </StackPanel>
+    </DataTemplate>
+</ResourceDictionary>
+
+<TextBox Validation.ErrorTemplate="{StaticResource ValidationErrorTemplate}"/>
+```
+
+## Change the resource color for the error and description messages
+Colors for the error and description messages are defined in a standard color/brush pattern in generic.xaml that can be overridden at an app level.
+
+```xaml
+<Application.Resources>
+
+    <!--Color for error messages-->
+    <Color x:Key="SystemErrorColor">#CE3880</Color>
+    <SolidColorBrush x:Key="SystemControlErrorTextForegroundBrush" Value="SystemErrorColor"/>
+
+    <!--Brush for description messages-->
+    <SolidColorBrush x:Key="SystemControlDescriptionTextForegroundBrush" Value="SystemControlPageTextBaseMediumBrush"/>
+
+</Application.Resources>
+```
+
+## Change description text for validation
```

Comment:
This seems like something that warrants a spec in its own right as it seems bigger than is covered here.
Is this displayed only when there's an error?
How to display it based on custom logic?
How to restyle this?
How can this be incorporated in custom controls?

Reply:
I agree, I'm not sure what this is doing in this spec.

Reply:
We will update this section to call out to use description with input validation and remove the overall description information.